### ISSUE TITLE
Eliminate ErrorProne warnings

### ISSUE THREAD

**Issue by chanseokoh** [GoogleContainerTools/jib#3359]
Maybe new warnings after upgrading ErrorProne in #3355.

```
/home/runner/work/jib/jib/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java:250: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
58

59
   * Converts a map of volumes strings to a set of {@link AbsoluteUnixPath}s (e.g. {@code {@code
60
> Task :jib-core:compileJava
61
                                                                                          ^
62
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
```

```
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpec.java:45: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?

 * from: see {@link BaseImageSpec}
> Task :jib-cli:compileJava
             ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpec.java:63: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 * layers: see {@link LayersSpec}
               ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BaseImageSpec.java:32: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - see {@link PlatformSpec}
           ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BaseImageSpec.java:33: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - see {@link PlatformSpec}
           ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/LayersSpec.java:31: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 * properties: see {@link FilePropertiesSpec}
                   ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/LayersSpec.java:33: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - see {@link LayerSpec}
           ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/LayersSpec.java:34: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - see {@link LayerSpec}
           ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/FileLayerSpec.java:35: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - {@link CopySpec}
       ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/FileLayerSpec.java:36: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 *   - {@link CopySpec}
       ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/FileLayerSpec.java:38: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 * properties: see {@link FilePropertiesSpec}
                   ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
/home/runner/work/jib/jib/jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/CopySpec.java:42: warning: [UnrecognisedJavadocTag] This Javadoc tag wasn't recognised by the parser. Is it malformed somehow, perhaps with mismatched braces?
 * properties: see {@link FilePropertiesSpec}
                   ^
    (see https://errorprone.info/bugpattern/UnrecognisedJavadocTag)
11 warnings
```

```
/home/runner/work/jib/jib/jib-maven-plugin/src/integration-test/java/com/google/cloud/tools/jib/maven/BuildImageMojoIntegrationTest.java:144: warning: [CatchFail] Ignoring exceptions and calling fail() is unnecessary, and makes test output less useful
228

229
    try {
230
> Task :jib-maven-plugin:compileIntegrationTestJava
231
    ^
232

233
    (see https://errorprone.info/bugpattern/CatchFail)
234
  Did you mean 'throw new AssertionError("error replacing tag with digest", ex);'?
235
```

The following one needs careful verification.
```
/home/runner/work/jib/jib/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java:332: warning: [UnnecessaryAssignment] Fields annotated with @Inject should not be manually assigned to, as they should be initialized by a framework. Remove the assignment if a framework is being used, or the annotation if one isn't.
209
  @Inject private Set<JibMavenPluginExtension<?>> injectedPluginExtensions = Collections.emptySet();
210
                                                  ^
211
    (see https://errorprone.info/bugpattern/UnnecessaryAssignment)
212
  Did you mean '@Inject private Set<JibMavenPluginExtension<?>> injectedPluginExtensions ;' or 'private Set<JibMavenPluginExtension<?>> injectedPluginExtensions = Collections.emptySet();'?
213

214
> Task :jib-maven-plugin:compileJava
215
```

**Comment by rainyShine** [comment]
Hello, I would like to contribute to this issue.
Could you assign this to me?

**Comment by chanseokoh** [comment]
@rainyShine sure, thanks for your interest! There may be more warnings than what I recorded here.

**Comment by rainyShine** [comment]
Thank you for assigning me.

I have come up with a solution for the part where the warning occurs in UnrecognizedJavadocTag and CatchFail.

- UnrecognizedJavadocTag part: Enclose each with {@code}.

- CatchFail part: Modify according to the following link.
[https://errorprone.info/bugpattern/CatchFail](https://errorprone.info/bugpattern/CatchFail)

**However, regarding the @Inject part, although there is a countermeasure, I can not judge whether it is correct.
Is it okay to create a setter and add @Inject to it?**

**Comment by chanseokoh** [comment]
Ah, I think we need to be careful about the `@Inject` warnings. Let's not touch that. I'll need to look into it later.

**Comment by meltsufin** [comment]
Fixed in #3410.
